### PR TITLE
Add MyPy support

### PR DESCRIPTION
**Pull request by kukushechkin**

### CODE REVIEW

File: .mypy.ini [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[mypy-mockito.*]
+ignore_missing_imports = True
```
Comment (hay-k) [inline .mypy.ini:2]:
```suggestion
ignore_missing_imports = True

```

File: src/iqm_client/iqm_client.py [hunk 2]
```diff
@@ -562,6 +566,7 @@ def submit_circuits(
             join(self._base_url, 'jobs'),
             json=data.dict(exclude_none=True),
             headers=headers,
+            timeout = 10
```
Comment (hay-k) [inline src/iqm_client/iqm_client.py:569]:
```suggestion
            timeout = REQUESTS_TIMEOUT
```

File: src/iqm_client/iqm_client.py [hunk 3]
```diff
@@ -586,16 +591,17 @@ def get_run(self, job_id: UUID) -> RunResult:
         bearer_token = self._get_bearer_token()
         result = requests.get(
             join(self._base_url, 'jobs/', str(job_id)),
-            headers=None if not bearer_token else {'Authorization': bearer_token}
+            headers=None if not bearer_token else {'Authorization': bearer_token},
+            timeout = 10
```
Comment (hay-k) [inline src/iqm_client/iqm_client.py:595]:
```suggestion
            timeout = REQUESTS_TIMEOUT
```

?

File: CHANGELOG.rst [hunk 4]
```diff
@@ -2,6 +2,11 @@
 Changelog
 =========
 
+Version 6.2
+===========
+
+* Add mypy support. `#41 <https://github.com/iqm-finland/iqm-client/pull/41>`_
```
Comment (hay-k) [inline CHANGELOG.rst:8]:
```suggestion
* Enable mypy checks. `#41 <https://github.com/iqm-finland/iqm-client/pull/41>`_
* Update source code according to new checks in pylint v2.15.0. `#41 <https://github.com/iqm-finland/iqm-client/pull/41>`_
```

File: src/iqm_client/iqm_client.py [hunk 5]
```diff
@@ -119,6 +119,8 @@
 import requests
 from pydantic import BaseModel, Field
 
+REQUESTS_TIMEOUT = 10
```
Comment (hay-k) [inline src/iqm_client/iqm_client.py:122]: What was the timeout before? I mean, we didn't specify it, but what was it by default?

Comment (kukushechkin) [reply src/iqm_client/iqm_client.py:122]: From [docs](https://requests.readthedocs.io/en/latest/user/advanced/#timeouts): "By default, requests do not time out unless a timeout value is set explicitly."

File: CHANGELOG.rst (hunk 4 above)
Comment (kukushechkin) [reply CHANGELOG.rst:8]: Technically, this PR will also allow users to use mypy on their code without "package 'iqm-client' is missing `py.typed` file" failure.

File: src/iqm_client/iqm_client.py (hunk 5 above)
Comment (hay-k) [reply src/iqm_client/iqm_client.py:122]: In that case we may want to expose this timeout to the end user for the calls which retrieve the results for a circuit run. The results might be huge (depending on the number of qubits and shots), and may not fully arrive within the default timeout. The user should have a way to override this

Comment (kukushechkin) [reply src/iqm_client/iqm_client.py:122]: Single `timeout` value is for both connect and read. It is a technical timeout value and is not the timeout for the whole transmission to finish. Read timeout is the max time between data arriving to the socket. Did some testing — with timeout=1 100000 shots for a circuit finishes in ~2 min.

So it looks like 10 seconds is enough and if it starts to timeout either for connect or between packets — this is a signal to investigate what is going on with connectivity.

**Review by hay-k — APPROVED** [review]

**Pull request merged**